src/spark_rapids_ml/knn.py

```diff
@@ -295,11 +276,66 @@ def _out_schema(self) -> Union[StructType, str]:
             ]
         )
 
-    def _get_cuml_fit_func(
+    def _require_ucx(self) -> bool:
+        """If enable or disable ucx over NCCL"""
+        return True
+
+    def _pre_process_data(  # type: ignore
+        self, dataset: DataFrame
+    ) -> Tuple[
+        List[Column], Optional[List[str]], int, Union[Type[FloatType], Type[DoubleType]]
+    ]:
+        (
+            select_cols,
+            multi_col_names,
+            dimension,
+            feature_type,
+        ) = super()._pre_process_data(dataset)
+
+        # if input format is vectorUDT, convert data type from float64 to float32
+        input_col, _ = self._get_input_columns()
+        if input_col is not None and isinstance(
+            dataset.schema[input_col].dataType, VectorUDT
+        ):
+            select_cols[0] = vector_to_array(col(input_col), dtype="float32").alias(
+                alias.data
+            )
+
+        select_cols.append(col(alias.label))
+
+        if self.hasParam("id_col") and self.isDefined("id_col"):
+            id_col_name = self.getOrDefault("id_col")
+            select_cols.append(col(id_col_name).alias(alias.row_number))
+        else:
+            select_cols.append(col(alias.row_number))
+
+        return select_cols, multi_col_names, dimension, feature_type
+
+    def kneighbors(self, query_df: DataFrame) -> Tuple[DataFrame, DataFrame, DataFrame]:
```

Comment:
Should add pydocs for this method since it's a new method, esp. explaining arguments and return values.

Reply:
done. Thanks for the advice!